src/main/scala/scala/Xgettext.scala

```diff
@@ -125,21 +131,30 @@ msgstr ""
               val line       = (relPath(pos.source.path), pos.line)
 
               if (i18n_t.contains(methodName)) {
-                val msgid = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                msgToLines.addBinding((None, msgid, None), line)
+                stringConstant(list.head, pos) match {
+                  case Some(msgid) =>
+                    msgToLines.addBinding((None, fixBackslashSingleQuote(msgid), None), line)
+                  case _ => // do nothing
+                }
               } else if (i18n_tn.contains(methodName)) {
-                val msgid       = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                val msgidPlural = fixBackslashSingleQuote(stringConstant(list(1), pos))
-                msgToLines.addBinding((None, msgid, Some(msgidPlural)), line)
+                (stringConstant(list.head, pos), stringConstant(list(1), pos)) match {
+                  case (Some(msgid), Some(msgidPlural)) =>
+                    msgToLines.addBinding((None, fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))), line)
+                  case _ => // do nothing
+                }
               } else if (i18n_tc.contains(methodName)) {
-                val msgctxt = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                val msgid   = fixBackslashSingleQuote(stringConstant(list(1), pos))
-                msgToLines.addBinding((Some(msgctxt), msgid, None), line)
+                (stringConstant(list.head, pos), stringConstant(list(1), pos)) match {
+                  case (Some(msgctxt), Some(msgid)) =>
+                    msgToLines.addBinding((Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid), None), line)
+                  case _ => // do nothing
+                }
               } else if (i18n_tcn.contains(methodName)) {
-                val msgctxt     = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                val msgid       = fixBackslashSingleQuote(stringConstant(list(1), pos))
-                val msgidPlural = fixBackslashSingleQuote(stringConstant(list(2), pos))
-                msgToLines.addBinding((Some(msgctxt), msgid, Some(msgidPlural)), line)
+                (stringConstant(list.head, pos), stringConstant(list(1), pos), stringConstant(list(2), pos)) match {
+                  case (Some(msgctxt), Some(msgid), Some(msgidPlural)) =>
+                    msgToLines.addBinding((Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid),
+                      Some(fixBackslashSingleQuote(msgidPlural))), line)
+                  case _ => // do nothing
```

Comment:
The `case _ => // do nothing` doesn't look very nice.

Let's use for comprehension, like this:

```
for (msgid <- stringConstant(list.head, pos)) {
  msgToLines.addBinding(
    (None, fixBackslashSingleQuote(msgid), None),
    line
  )
}

for (msgid <- stringConstant(list.head, pos); msgidPlural <- stringConstant(list(1), pos)) {
  msgToLines.addBinding(
    (None, fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))),
    line
  )
}

for (msgctxt <- stringConstant(list.head, pos); msgid <- stringConstant(list(1), pos)) {
  msgToLines.addBinding(
    (Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid), None),
    line
  )
}

for (msgctxt <- stringConstant(list.head, pos); msgid <- stringConstant(list(1), pos); msgidPlural <- stringConstant(list(2), pos)) {
  msgToLines.addBinding(
    (Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))),
    line
  )
}
```

Reply:
Right, updated my PR.